.github/workflows/cryostat-IT.yml

```diff
@@ -0,0 +1,119 @@
+on:
+  workflow_call:
+    inputs:
+      build-arch:
+        required: true
+        type: string
+      skip-itests:
+        required: false
+        type: boolean
+        default: false
+    outputs:
+      image-version:
+        description: the Cryostat application version that will be built
+        value: ${{ jobs.get-pom-properties.outputs.image-version }}
+
+permissions:
+  contents: read
+  packages: read
+
+jobs:
+  get-pom-properties:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: cryostatio/cryostat
+        ref: main
+    - id: query-pom
+      name: Get properties from POM
+      # Query POM for image version and save as output parameter
+      run: |
+        IMAGE_VERSION="$(mvn validate help:evaluate -Dexpression=cryostat.imageVersionLower -q -DforceStdout)"
+        echo "image-version=$IMAGE_VERSION" >> "$GITHUB_OUTPUT"
+    outputs:
+      image-version: ${{ steps.query-pom.outputs.image-version }}
+
+  build-image:
+    runs-on: ubuntu-latest
+    needs: [get-pom-properties]
+    steps:
+    - name: Install qemu
+      if: ${{ inputs.build-arch != 'amd64' }}
+      continue-on-error: true
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y qemu-user-static
+    - uses: actions/checkout@v4
+      if: always()
+      with:
+        repository: cryostatio/cryostat
+        ref: main
+        submodules: true
+        fetch-depth: 0
+    - uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+    - name: maven-settings
+      uses: s4u/maven-settings-action@v2
+      with:
+        servers: '[{"id": "github", "username": "dummy", "password": "${env.GITHUB_TOKEN_REF}"}]'
+        githubServer: false
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: restore
+    - run: git submodule init
+    - run: git submodule update
+    - run: mvn -B -U -Dbuild.arch=${{ inputs.build-arch }} clean package
```

Comment:
Does this image use the `-core` version built from `push/PR` commit?

Reply:
I think what we can do is cache the artifact from `build` job in `.github/workflows/ci.yaml`. Then, restore it here and do an offline build here.

Reply:
Will need to patch the `pom.xml` on cryostat side tho.